proxy/src/http/source.rs

```diff
@@ -318,18 +306,17 @@ mod handler {
         }: super::BlobQuery,
     ) -> Result<impl Reply, Rejection>
     where
-        R: registry::Client,
+        R: registry::Client + 'static,
     {
-        let registry = registry.read().await;
-        let store = store.read().await;
-        let session = session::current(Arc::clone(&api), &*registry, &store).await?;
+        let ctx = ctx.read().await;
+
+        let session = session::current(&ctx.peer_api, &ctx.registry, &ctx.store).await?;
```

Comment:
I think it might make more sense for us to pass the theme down into this function.

Reply:
You mean into the call to `session::current`? The theme comes from the settings which stored in the session. A bit confused what you proposing.

Reply:
Well, the thing was that I started asking myself why we needed a `Registry` here and then I realised we needed it for the session, which in turn made me wonder why needed the session. But really, what we need is a `theme`. Maybe it'd be better as a follow-up ticket in a similar vein to what I said about `with_peer_api` :)